### DISCUSSION TITLE
How to set 'sma'

### DISCUSSION THREAD

**Discussion by Yzxastr** [phoebe-project/phoebe2#649] [Questions]
As we know, the 'sma' is an absolute parameters for a  binary , not only by the light curves to solve, but we can only from the light curve solve the 'sma' when we use the phoebe optimizer . So I would like to ask how the optimizer can solve for the 'sma'  only from the light curve, and how do I set the 'sma' when using the optimizer to solve for other relative parameters.

**Comment by kecnry** [comment]
(I moved this to a discussion rather than an issue so its easier for others to search/find).  

The optimizers will just attempt to fit any parameters you ask them to (via the `fit_parameters` parameter) by optimizing the cost function between the observations and the model for all enabled datasets while leaving all other parameters fixed at their face values.  It is therefore up to you to come up with a "strategy" for optimizing appropriate parameters given the type of system and observations that you have.  If I have an SB2 system with RVs, for example, I like to first optimize just the RVs for asini, q, and perhaps eccentricity, and then leave asini fixed and fit the light curves for inclination and then slowly introduce more parameters.  This case also requires [flipping constraints](http://phoebe-project.org/docs/latest/tutorials/constraints) in order to change the parameterization for asini, sma, and incl.

I'm not sure if this fully answered your question - so let me know if I missed something and we'll try to help.

**Comment by Yzxastr** [comment]
First of all, thank you for your answer, but my main question is how the optimizer can calculate the sma without the RVs. Is it approximated by something or is it constrained by something else?




---Original---
From: "Kyle ***@***.***&gt;
Date: Wed, Nov 2, 2022 21:42 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [phoebe-project/phoebe2] How to set 'sma' (Discussion #649)




 
(I moved this to a discussion rather than an issue so its easier for others to search/find).
 
The optimizers will just attempt to fit any parameters you ask them to (via the fit_parameters parameter) by optimizing the cost function between the observations and the model for all enabled datasets while leaving all other parameters fixed at their face values.  It is therefore up to you to come up with a "strategy" for optimizing appropriate parameters given the type of system and observations that you have.  If I have an SB2 system with RVs, for example, I like to first optimize just the RVs for asini, q, and perhaps eccentricity, and then leave asini fixed and fit the light curves for inclination and then slowly introduce more parameters.  This case also requires flipping constraints in order to change the parameterization for asini, sma, and incl.
 
I'm not sure if this fully answered your question - so let me know if I missed something and we'll try to help.

**Reply by kecnry** [reply]
The optimizers don't calculate anything, they vary the parameters that you tell them to fit, run the forward model, and then compare that to the observations.  If the parameter has no effect on the model at all, then the results from the optimization would be essentially random (whereas a sampler would show this degeneracy).  Depending on your systems and other parameters, sma might have some influence on the light curves though (by changing the distortion if you have a fixed star radius, for example).

**Comment by Yzxastr** [comment]
Thanks a lot ,I think I know what I'm looking for.



---Original---
From: "Kyle ***@***.***&gt;
Date: Thu, Nov 3, 2022 00:22 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [phoebe-project/phoebe2] How to set 'sma' (Discussion #649)




 
The optimizers don't calculate anything, they vary the parameters that you tell them to fit, run the forward model, and then compare that to the observations.  If the parameter has no effect on the model at all, then the results from the optimization would be essentially random (whereas a sampler would show this degeneracy).  Depending on your systems and other parameters, sma might have some influence on the light curves though (by changing the distortion if you have a fixed star radius, for example).
